cmd/zdap/commands/clone.go

```diff
@@ -324,7 +356,10 @@ func AttachClone(c *cli.Context) error {
 
 	if c.Bool("new") {
 		fmt.Print("Cloning ", resource, "...")
-		clone, err = cloneResource(c.Args().Slice())
+		clone, err = cloneResource(c.Args().Slice(), zdap.ClaimArgs{
```

Comment:
If we should support to attach to a claimed clone (`zdap claim ...` command), we should pass the same `zdap.ClaimArgs` to `findClone` below as well (when `c.Bool("new")` is false.

Reply:
I'm leaning towards not supporting attaching to a claim

Reply:
One issue I'm seeing with supporting attaching to a claim as if it were a regular clone is using the timestamp as identification. It doesn't seem entirely unlikely that two clones might get the same created_at if we start using this more aggressively in automated tests. The "clone_id" we return when claiming a pooled clone has the format `zdap-postgres-trade-base-2023-12-12T11.32.00-clone-2024-02-22T08.19.25.TsF` and should be more robust.

Reply:
What's the use-case for `zdap claim` command? If we don't really need that, this is a non-issue since we'll always use `zdap attach` command to get a fresh clone from the pool. But if we have use-cases where we need the `zdap claim` command, we should make sure that we can attach to it also.

Reply:
I've used the binary, then I guess an assumption would be that anyone who runs a test like that has an up to date zdap installed and in their path. You're thinking we should then call the zdap servers directly? Having zdap as a dependency when it's in go?

Reply:
Yes, it's easier to handle structs/JSON than parse std out from a cli-program, and I guess that we could build client's for other languages as well if we want to. I'm not saying that it's something that we should focus on right now, but with the pool feature, I think it would be really good to be able to integrate zdap into the test suite as a dependency, but that requires us to refactor `cmd/zdap` a bit, and mainly split the functions in `cmd/zdap/commands` into a `client` package, so it's easy to use as a dependency.

Reply:
Yeah I agree that makes sense